src/olympia/versions/models.py

```diff
@@ -625,19 +630,20 @@ def clear_compatversion_cache_on_delete(sender, instance, **kw):
 models.signals.post_save.connect(
     inherit_nomination, sender=Version,
     dispatch_uid='version_inherit_nomination')
-models.signals.post_delete.connect(
-    update_status, sender=Version, dispatch_uid='version_update_status')
 models.signals.post_save.connect(
     update_incompatible_versions, sender=Version,
     dispatch_uid='version_update_incompat')
-models.signals.post_delete.connect(
-    update_incompatible_versions, sender=Version,
-    dispatch_uid='version_update_incompat')
-models.signals.pre_delete.connect(
-    cleanup_version, sender=Version, dispatch_uid='cleanup_version')
 models.signals.post_save.connect(
     clear_compatversion_cache_on_save, sender=Version,
     dispatch_uid='clear_compatversion_cache_save')
+
+models.signals.pre_delete.connect(
+    cleanup_version, sender=Version, dispatch_uid='cleanup_version')
+models.signals.post_delete.connect(
+    update_status, sender=Version, dispatch_uid='version_update_status')
+models.signals.post_delete.connect(
+    update_incompatible_versions, sender=Version,
+    dispatch_uid='version_update_incompat')
 models.signals.post_delete.connect(
```

Comment:
This was just me re-organising to make it more understandable (for me). If it was already in a defined order can revert.